src/hmem_cuda_gdrcopy.c

```diff
@@ -292,6 +293,44 @@ void cuda_gdrcopy_from_dev(uint64_t handle, void *hostptr,
 			  GDRCOPY_FROM_DEVICE);
 }
 
+static ssize_t cuda_gdrcopy_iov_buf(uint64_t handle,
+                                    const struct iovec *iov, size_t iov_count,
+                                    uint64_t iov_offset, void *buf,
+                                    size_t size, enum gdrcopy_dir dir)
+{
+	uint64_t done = 0, len;
+	char *dev_buf;
+	size_t i;
+
+	for (i = 0; i < iov_count && size; i++) {
+		len = ofi_iov_bytes_to_copy(&iov[i], &size, &iov_offset, &dev_buf);
+
+		if (!len)
+			continue;
+
+		cuda_gdrcopy_impl(handle, dev_buf, (void *)((char *)buf + done), len, dir);
+
+		done += len;
+	}
+	return done;
+}
+
+ssize_t ofi_gdrcopy_to_cuda_iov(uint64_t handle, const struct iovec *iov,
```

Comment:
The naming doesn't seem to align with how the rest of the functions are named, perhaps it should be called
```ofi_cuda_gdrcopy_to_iov```? This is a minor NIT, feel free to ignore it if you'd like.

Reply:
The way I look at it is this symmetry
```
ofi_copy_from_hmem_iov
ofi_gdrcopy_from_cuda_iov
```

On the other hand `ofi_cuda_gdrcopy_to_iov` is vague on whether the iov is on host or cuda.